components/src/AddressForm/addressform.main.tsx

```diff
@@ -139,120 +130,86 @@ class AddressFormMain extends React.Component<AddressFormMainProps, AddressFormM
     this.setState({ postalCode: event.target.value });
   }
 
-  submitAddress(event) {
+  async submitAddress(event) {
     event.preventDefault();
     const { addressData, fetchData, onCloseModal } = this.props;
     const {
-      addressForm, firstName, lastName, address, extendedAddress, city, country, subCountry, postalCode,
+      firstName, lastName, address, extendedAddress, city, country, subCountry, postalCode,
     } = this.state;
-    let link;
-    let methodType;
+    const userAddress = {
+      name: {
+        'given-name': firstName,
+        'family-name': lastName,
+      },
+      address: {
+        'street-address': address,
+        'extended-address': extendedAddress,
+        locality: city,
+        'country-name': country,
+        region: subCountry,
+        'postal-code': postalCode,
+      },
+    };
+    let request;
     if (addressData && addressData.address) {
-      link = addressData.address;
-      methodType = 'put';
+      request = this.client.address(addressData.address).update(userAddress);
```

Comment:
It's better to use `await` for all `fetch/update` calls for better readability and consistency. Especially since this is inside `async` method already.

Reply:
Thanks for the review!